demisto_sdk/commands/lint/resources/pylint_plugins/partner_level_checker.py

```diff
@@ -106,6 +111,20 @@ def _init_params_checker(self, node):
         except AttributeError:
             pass
 
+    def _init_args_checker(self, node):
+        try:
+            if node.func.attrname == 'args' and node.func.expr.name == 'demisto':
+                check_param = True
+                parent = node.parent
+                while check_param and parent:
+                    if isinstance(parent, astroid.FunctionDef) and parent.name == 'main':
+                        check_param = False
+                    parent = parent.parent
+                if check_param:
+                    self.add_message("init-args-outside-main", node=node)
+        except AttributeError:
+            pass
```

Comment:
I'm do not remember demisto.args() should be only in main.  But maybe my memory is misleading me. @ShahafBenYakir / @yaakovi ?

Reply:
I asked @guyfreund , according to him it should.

Reply:
Nope this wasn't a required check

Reply:
It's better to initialize it in the main and pass it to the other functions to make UT easier, this of course should be presented to the user as a warning.

Reply:
@guyfreund at what point did we say this?

Reply:
If you'll call demisto.args in a function you'll have to mock it, if you pass it as an argument then no need for the mocking.

This might not be enforced but it's better practice for our xsoar supported integrations.

Reply:
maybe we should put this not as a W but as a Convention, @orhovy ?

Reply:
I agree with @guyfreund, there should be a single call to `demisto.args()` (and `demisto.params()` for integrations) at the main or prior to it.
this is the best practice, but since most of our code doesn't follow that guideline I would recommend keeping it as non-mandatory.

Reply:
After a shared call we agreed to leave this as warning